### ISSUE TITLE
tx_client: rework EstimateGas

### ISSUE THREAD

**Issue by vgonkivs** [celestiaorg/celestia-app#3524]
As we've added the option of auto- gas and fee calculations we should reconsider the estimation process for all transactions(besides `MsgPayForBlob`) and get rid of the `Simulate` request as it brings an additional overhead in the form of roundtrips. EstimateGas should work the same as it works for [PFB transactions](https://github.com/celestiaorg/celestia-app/blob/af426bf8062247af817f13ed82c66851213513e2/x/blob/types/payforblob.go#L171). To achieve this, we should empirically get constants for the estimation. I'd also propose breaking the issue into multiple PRs, so that each of them will contain the estimation for different tx types(I believe it will help to speed up the implementation).

**Comment by cmwaters** [comment]
I don't think we would ever get rid of estimate gas. There are so many message types (you can also mix and match), I don't think it's worth building a model for each but rather just for the most frequently used message types.

**Comment by vgonkivs** [comment]
Yes, we can start with MsgSend/MsgDelegate and then see how it goes 🙂. But, I'd personally, get rid of `Simulate`.
